pgmpy/inference/CausalInference.py

```diff
@@ -986,6 +986,15 @@ def query(
                 "`evidence` must be a dict of the form: {variable_name: variable_state}"
             )
 
+        if do:
+            for var in variables:
+                for do_var in do.keys():
```

Comment:
We can simplify this nested loop using `itertools.product`. We can replace it with: `for var, do_var in itertools.product(variables, do):`